Hazel/src/Platform/Windows/WindowsWindow.cpp

```diff
@@ -172,5 +178,46 @@ namespace Hazel {
 	{
 		return m_Data.VSync;
 	}
+	
+	void WindowsWindow::SetWindowMode(const WindowMode& mode)
+	{
+		HZ_ASSERT(m_Window, "Failed to retrieve window.");
+		if (mode == m_Data.Mode) return;
+
+		unsigned int width = 0, height = 0;
+
+		m_PrimaryMonitor = glfwGetPrimaryMonitor();
+		m_BaseVideoMode = *(glfwGetVideoMode(m_PrimaryMonitor));
+
+		GLFWmonitor* monitor = nullptr;
+
+		if (mode == WindowMode::Windowed) {
+			if (width == 0 || height == 0) {
+				width = m_Data.Width;
+				height = m_Data.Height;
+			}
+			glfwGetWindowPos(m_Window, &(m_Data.XPos), &(m_Data.YPos));
+		}
+		else if (mode == WindowMode::Borderless) {
+			width = m_BaseVideoMode.width;
+			height = m_BaseVideoMode.height;
+			monitor = m_PrimaryMonitor;
+		}
+		else if (mode == WindowMode::FullScreen) {
+			if (width == 0 || height == 0) {
+				// TODO: Change this to check if it is a valid full screen resolution pair
+				width = m_Data.Width;
+				height = m_Data.Height;
```

Comment:
I just want to have notion of this in a review that this is not ready to merge, as this needs to be fixed still. I am aware this is still a todo.